src/unix/fs.c

```diff
@@ -457,7 +457,7 @@ static ssize_t uv__fs_preadv(uv_file fd,
 
 
 static ssize_t uv__fs_read(uv_fs_t* req) {
-#if defined(__linux__)
+#if defined(__linux__) && !(defined(__ANDROID__) && __ANDROID_API__ < 24)
   static _Atomic int no_preadv;
 #endif
```

Comment:
Can you add a comment as in the `getgrgid_r` case? Also, could we just initialize the `no_preadv` to `1` in case `__ANDROID_API__ < 24` and be done with it?

Reply:
I tried it, but it still doesn't work after compiling. I added some comments.